flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1776,6 +1776,25 @@ public ResourceID getResourceID() {
 		return unresolvedTaskManagerLocation.getResourceID();
 	}
 
+	public long getUsedManagedMemory() {
```

Comment:
I understand what you mean saying that the `TaskExecutor` is already kind of busy doing other stuff. I'm just not sure whether it makes sense to move the logic somewhere else. It is actually a property of `TaskExecutor`. Alternatively, we could think about moving the logic into `TaskSlotTable`. Semantically, it would fit in there even more. Instead of `MetricUtils.instantiateManagedMemoryMetrics(MetricGroup, TaskExecutor)` we could use `MetricUtils.instantiateManagedMemoryMetrics(MetricGroup, TaskSlotTable)` instead. What do you think about that?

I would object moving the aggregating code into `MetricUtils`. The sole purpose of `MetricUtils` seem to be that it instantiates metrics. There is no other code. Hence, I'd like to keep it that way.

Reply:
We agreed offline to move it as it would make the code cleaner. Additionally, it doesn't add more complexity to the `TaskExecutor` class as @azagrebin already mentioned.